ambry-network/src/test/java/com.github.ambry.network/NetworkClientTest.java

```diff
@@ -313,6 +316,51 @@ public void testConnectionInitializationFailures() {
     responseInfoList.clear();
   }
 
+  /**
+   * Test that connections get replenished in {@link NetworkClient#sendAndPoll(List, int)} to maintain the minimum
+   * number of active connections.
+   */
+  @Test
+  public void testConnectionReplenishment() {
+    AtomicInteger nextCorrelationId = new AtomicInteger(1);
+    Function<Integer, List<RequestInfo>> requestGen = numRequests -> IntStream.range(0, numRequests)
+        .mapToObj(
+            i -> new RequestInfo(sslHost, sslPort, new MockSend(nextCorrelationId.getAndIncrement()), replicaOnSslNode))
+        .collect(Collectors.toList());
+    // 1 host x 1 port x 3 connections x 100%
+    AtomicInteger expectedConnectCalls = new AtomicInteger(3);
+    Runnable checkConnectCalls = () -> Assert.assertEquals(expectedConnectCalls.get(), selector.connectCallCount());
+
+    networkClient.warmUpConnections(Collections.singletonList(replicaOnSslNode.getDataNodeId()), 100, 2000,
```

Comment:
replace '100' and '2000' with static final ints?

Reply:
replaced the poll timeout and warmup timeout with a constant.

The warmup % has a different meaning than poll timeout, so I made it a local variable